### ISSUE TITLE
Math seems to override JSON-set spell maxes

### ISSUE THREAD

**Issue by Standing-Storm** [CleverRaven/Cataclysm-DDA#67589]
### Describe the bug

Spells set to dynamically scale using math seem to be able to exceed limits set by the max_[whatever] JSON entries.

### Attach save file

[Mindenmines-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/12317415/Mindenmines-trimmed.tar.gz)

### Steps to reproduce

1) Load game
2) Try to use Flamethrower
3) Notice the Flamethrower power lets you blow up the entire screen

Flamethrower has "max_aoe": 180 and "max_range": 20 set in JSON, but the min_range and min_aoe are set with math statements and seem to scale infinitely.

Flamethrower JSON located here: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/MindOverMatter/powers/pyrokinesis.json

### Expected behavior

Math statements are still capped by the max values listed in the spell's JSON

### Screenshots

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/120433252/731a4252-fb13-4e38-93fe-33ad99ae623e)

![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/120433252/e0385891-7b87-463b-a217-caf82d06304d)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3324 (22H2)
- Game Version: d044f5f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    No Rail Stations [No_Rail_Stations],
    Mind Over Matter [mindovermatter]
]

**Comment by MoonlitRiverlight** [comment]
Additional detail on this: Get high enough on levels (86 in my case) and the spells wrap around to their minimum effect/range/duration values.

**Comment by andrei8l** [comment]
This is not a math issue and you can reproduce it with static numbers. The aoe code is just a little kooky. Probably the intention was this:

```diff
diff --git a/src/magic.cpp b/src/magic.cpp
index aed5b1f723..2a7e5ae885 100644
--- a/src/magic.cpp
+++ b/src/magic.cpp
@@ -840,11 +840,11 @@ int spell::aoe( const Creature &caster ) const
         return rng( std::min( leveled_aoe, static_cast<int>( type->max_aoe.evaluate( d ) ) ),
                     std::max( leveled_aoe, static_cast<int>( type->max_aoe.evaluate( d ) ) ) );
     } else {
-        if( type->max_aoe.evaluate( d ) >= type->min_aoe.evaluate( d ) ) {
-            return std::min( leveled_aoe, static_cast<int>( type->max_aoe.evaluate( d ) ) );
-        } else {
-            return std::max( leveled_aoe, static_cast<int>( type->max_aoe.evaluate( d ) ) );
+        if( type->max_aoe.evaluate( d ) > type->min_aoe.evaluate( d ) ) {
+            return std::clamp( leveled_aoe, static_cast<int>( type->min_aoe.evaluate( d ) ),
+                               static_cast<int>( type->max_aoe.evaluate( d ) ) );
         }
+        return std::min( leveled_aoe, static_cast<int>( type->max_aoe.evaluate( d ) ) );
     }
 }
 
```

Note that this favours `max_aoe` over `min_aoe` and I bet someone will eventually consider THAT a bug. It needs to be documented.

**Comment by Standing-Storm** [comment]
It's not just AoE, it occurs with range as well--the range in the screenshot above is 48, despite the [max_range being 20 in the code.](https://github.com/CleverRaven/Cataclysm-DDA/blob/264a86adc355354f22f3ad2dc450b29453c73d2e/data/mods/MindOverMatter/powers/pyrokinesis.json#L215C5-L215C21)

**Comment by andrei8l** [comment]
Yes, the same if-else code is used for range, duration, damage, accuracy, etc. It would have to be fixed the same way in all the spots.
